CMakeLists.txt

```diff
@@ -219,6 +219,17 @@ target_link_libraries(${NVFUSER_CODEGEN} PRIVATE stdc++fs)
 target_link_libraries(${NVFUSER_CODEGEN} PRIVATE ${CUDA_NVRTC_LIB} ${LIBNVTOOLSEXT})
 target_include_directories(${NVFUSER_CODEGEN} PRIVATE ${CUDA_INCLUDE_DIRS})
 
+if (NOT UCC_FOUND)
+  # This probably means a "standalone" build, where UCC_FOUND is not inherited from pytorch's CMake configuration.
+  # It's required to set env vars UCC_DIR, UCX_DIR, UCC_HOME, UCX_HOME for CMake's find UCC to work.
+  find_package(UCC REQUIRED)
+  find_package(UCX REQUIRED)
+endif()
+add_library(__nvfuser_ucc INTERFACE)
+target_link_libraries(__nvfuser_ucc INTERFACE ucx::ucs ucx::ucp ucc::ucc)
+target_include_directories(__nvfuser_ucc INTERFACE ${UCC_INCLUDE_DIRS})
+target_link_libraries(${NVFUSER_CODEGEN} PRIVATE __nvfuser_ucc)
```

Comment:
I think you would need to check if PyTorch is built with UCC, and only do so if PyTorch is.

Reply:
dumb question, is pytorch exposing USE_UCC in config file? How are we checking that on a pip installed torch?

Reply:
On the python level, yes: https://github.com/pytorch/pytorch/blob/1a6619a830ad0fa7e364f1a4b178daf3d78b6e4c/test/run_test.py#L408

On CMake level, I don't know...